File: 0455-assign-cookies/0455-assign-cookies.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(s);
        Arrays.sort(g);

        int right=0;
        int satisfied=0;

       for(int left=0; left<s.length; left++){
            if(right == g.length)
            break;

            if(g[right] <= s[left]){
            satisfied++;
            right++;
        }

    }
         return satisfied;
    }
}

File: 0455-assign-cookies/0455-assign-cookies.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());

        int child = 0;

        for (int cookie : s) {
            if (child < g.size() && cookie >= g[child])
                ++child;
        }

        return child;
    }
};